release/src/router/www/client_function.js

```diff
@@ -3263,7 +3263,18 @@ function oui_query_web(mac){
 function clientFromIP(ip) {
 	for(var i=0; i<clientList.length;i++){
 		var clientObj = clientList[clientList[i]];
-		if(clientObj.ip == ip) return clientObj;
+		if(clientObj.ip == ip){
+			return clientObj;
+		}
+	}
+	staticList = originData.staticList;
+	for(var i=0; i<staticList.length; i++){
+		if(staticList[i] != ""){
+			data = staticList[i].split(">");
```

Comment:
Thoughts on commenting here (or better yet, [here](https://github.com/RMerl/asuswrt-merlin/blob/3f0224c32d4046d015d0f85cc6cc1b8acd278f2b/release/src/router/www/client_function.js#L227)?) what the expected format is? I had to go scrounging to find that `data[0]` is [0] the MAC address and that clientList is indeed keyed by MAC address.

How about: [1]

```
/* Entries in dhcp_staticlist are formatted as: "MAC>IP>hostname". */
staticList: decodeURIComponent('<% nvram_char_to_ascii("", "dhcp_staticlist"); %>').replace(/&#62/g, ">").replace(/&#60/g, "<").split('<'),
```

It could also be nice (though I don't feel too strongly about it) to set `var static_ip` and `var static_mac` or something from `data` instead of using them inline and leaving their values more implicit.

Other than that this LGTM.

-----

[0] Probably? I was able to find [`nvram_char_to_ascii`]((https://github.com/RMerl/asuswrt-merlin/blob/master/release/src/router/httpd/web.c#L970)) but not the implementation that reads from nvram. Is it in a binary somewhere? It seems like the larger need is documentation of what's in nvram and in what format, but I haven't run into that yet.

I was able to trace thusly:

* [`nvram_char_to_ascii`](https://github.com/RMerl/asuswrt-merlin/blob/3f0224c32d4046d015d0f85cc6cc1b8acd278f2b/release/src/router/www/client_function.js#L227) call
* [`nvram_char_to_ascii`](https://github.com/RMerl/asuswrt-merlin/blob/2ade8289eea4ea86489b02cc0cc638984ac9b95c/release/src/router/httpd/web.c#L15142) function association
* [`ej_nvram_char_to_ascii`](https://github.com/RMerl/asuswrt-merlin/blob/master/release/src/router/httpd/web.c#L970) definition
* [`nvram_safe_get_x`](https://github.com/RMerl/asuswrt-merlin/blob/master/release/src/router/httpd/web.c#L982) call
* [`nvram_safe_get_x`](https://github.com/RMerl/asuswrt-merlin/blob/master/release/src/router/httpd/bcmnvram_f.h#L49) definition / `nvram_get_x` call

but found no definition for `nvram_get_x()`.

[1] assuming the nvram value matches that in [`index.asp`](https://github.com/RMerl/asuswrt-merlin/blob/master/release/src/router/www/index.asp#L980) - the behavior of which suggests this is dumped straight into nvram which seems incredibly fragile.